### ISSUE TITLE
can import mituba inside blender?

### ISSUE THREAD

**Issue by zebus3d** [mitsuba-renderer/mitsuba2#139]
ImportError: The 'mitsuba' native modules could not be imported. You're likely trying to use Mitsuba within a Python binary (/Applications/Blender_2.90.app/Contents/MacOS/Blender) that is different from the one for which the native module was compiled (/Applications/Blender_2.90.app/Contents/Resources/2.90/python/bin/python3.7m).

Isn't it possible to import mitsuba python binding module from a addon in python for blender?.

Thx.

**Comment by bathal1** [comment]
Hi! 
Since Blender comes with its own bundled version of python, you need to compile mitsuba using the same python version as the one Blender uses:

- Figure out the version of python that your version of blender uses (python 3.74 for Blender 2.82, I don't know about 2.90)
- Recompile mitsuba using this version of python, either using a virtual environment, or by specifying the path to this python when running cmake with the option `-DPYTHON_EXECUTABLE=/path/to/other/python`

**Comment by zebus3d** [comment]
I've compiled with the same version as blender but I still get the same message, maybe I'm not doing something right, but I could check that in the __init__.py I have the following variables:

PYTHON_EXECUTABLE: /Applications/last_blender/Blender_2.90.app/Contents/Resources/2.90/python/bin/python3.7m
sys.executable: /Applications/last_blender/Blender_2.90.app/Contents/MacOS/Blender

I'm doing the import like this:
import Mitsuba.dist.python.mitsuba

Mitsuba is my folder of my blender addon. 
I've copied dist in.

**Comment by bathal1** [comment]
Copying the `dist/` folder is not the way to go to import the mitsuba python lib. Instead, you should append the path to `mitsuba/build/dist/python` to python's `sys.path`:
```python
import sys
sys.path.append("/path/to/mitsuba2/build/dist/python")

import mitsuba
```

It may also be preferable not to compile mitsuba 2 with blender's bundled python, since you have no control over its modules. For example, you won't be able to build mitsuba2 with sphinx or pytest since they're not bundled in blender's python. I would recommend you install python with the correct version and modules on your system, using a virtual environment, and use this python for compilation.

**Comment by zebus3d** [comment]
The first time I compiled it with this version: 
![Captura de pantalla 2020-05-25 a las 11 24 47](https://user-images.githubusercontent.com/48990185/82799637-a9c9e100-9e7a-11ea-971a-20f4db6a66d8.png)
But he told me it had to be the same version as Blender. Blender's is this one: 
![Captura de pantalla 2020-05-25 a las 11 26 07](https://user-images.githubusercontent.com/48990185/82799719-c403bf00-9e7a-11ea-8336-0c1137f1ab20.png)

So in the end I decided to try to compile with the same exact one to see if he kept telling me. 

Now when I have some time I'll try the sys path to import it and see how it goes. Thank you very much.
